src/components/ModalWithForm.js

```diff
@@ -26,6 +26,17 @@ export class ModalWithForm extends Modal {
     super.setEventListeners();
     this._form.addEventListener("submit", this._submitEventHandler);
   }
+
+  changeEventListener(callback) {
+    this._submitEventHandler = callback;
+    this.setEventListeners();
+    // this._form.removeEventListener("submit", this._submitEventHandler);
+    // this._form.addEventListener("submit", callback);
+  }
+
+  eliminateEventListener() {
+    this._form.removeEventListener("submit", this._submitEventHandler);
+  }
 }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`